src/x509_crl.ts

```diff
@@ -0,0 +1,309 @@
+
+    // Convert public key to CryptoKey
+    let publicKey: CryptoKey;
+    const paramsKey = params.publicKey;
+    try {
+      if (!paramsKey) {
+        throw new Error("Cannot get publicKey for verify sign");
+      } else if (paramsKey instanceof X509Certificate) {
+        // X509Certificate
+        keyAlgorithm = {
+          ...paramsKey.signatureAlgorithm,
+          ...paramsKey.signature,
+        };
+        publicKey = await paramsKey.publicKey.export(keyAlgorithm, ["verify"]);
+      } else if (paramsKey instanceof PublicKey) {
+        // PublicKey
+        keyAlgorithm = { ...paramsKey.algorithm, ...this.signature };
+        publicKey = await paramsKey.export(keyAlgorithm, ["verify"]);
+      } else {
+        // CryptoKey
+        keyAlgorithm = { ...paramsKey.algorithm, ...this.signature };
+        publicKey = paramsKey;
+      }
+    } catch (e) {
+      // NOTE: Uncomment the next line to see more information about errors
+      // console.error(e);
+
+      // Application will throw exception if public key algorithm is not the same type which is needed for
+      // signature validation (eg leaf certificate is signed with RSA mechanism, public key is ECDSA)
+      return false;
+    }
+
+    // Convert ASN.1 signature to WebCrypto format
+    const signatureFormatters = container.resolveAll<IAsnSignatureFormatter>(diAsnSignatureFormatter).reverse();
+    let signature: ArrayBuffer | null = null;
+    for (const signatureFormatter of signatureFormatters) {
+      signature = signatureFormatter.toWebSignature(keyAlgorithm, this.signature);
+      if (signature) {
+        break;
+      }
+    }
+    if (!signature) {
+      throw Error("Cannot convert ASN.1 signature value to WebCrypto format");
+    }
+
+    return await crypto.subtle.verify(this.signatureAlgorithm, publicKey, signature, this.tbs);
+  }
+
+  /**
+   * Returns a SHA-1 certificate thumbprint
+   * @param crypto Crypto provider. Default is from CryptoProvider
+   */
+  public async getThumbprint(crypto?: Crypto): Promise<ArrayBuffer>;
+  /**
+   * Returns a certificate thumbprint for specified mechanism
+   * @param algorithm Hash algorithm
+   * @param crypto Crypto provider. Default is from CryptoProvider
+   */
+  public async getThumbprint(
+    algorithm: globalThis.AlgorithmIdentifier,
+    crypto?: Crypto
+  ): Promise<ArrayBuffer>;
+  public async getThumbprint(...args: any[]) {
+    let crypto: Crypto;
+    let algorithm = "SHA-1";
+    if (args[0]) {
+      if (!args[0].subtle) {
+        // crypto?
+        algorithm = args[0] || algorithm;
+        crypto = args[1];
+      } else {
+        crypto = args[0];
+      }
+    }
+    crypto ??= cryptoProvider.get();
+
+    return await crypto.subtle.digest(algorithm, this.rawData);
+  }
+
+  /**
+   *  Gets the CRL entry, with the given X509Certificate or certificate serialNumber.
+   *
+   * @param {(...[certificate: X509Certificate] | [serialNumber: string])} args certificate | serialNumber
```

Comment:
Please use the same jsdoc. Don't add type declarations to jsdoc, TS does it

Looks like you are using VSCode extension which generates extended jsdoc comments